### ISSUE TITLE
初始化模型卡住

### ISSUE THREAD

**Issue by jinzhensheng** [verl-project/verl#3445]
### System Info

Verl==0.5.0

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

#!/bin/bash

 set -x

nohup python3 -u -m verl.trainer.main_ppo \
    algorithm.adv_estimator=grpo \
    algorithm.use_kl_in_reward=False \
    data.train_files=/data1/train_grpo.parquet \
    data.val_files=/data1/test_grpo.parquet \
    data.tokenizer=/data2/models/qwen2-7b-instruct \
    data.train_batch_size=2 \
    data.val_batch_size=1 \
    data.max_prompt_length=24760 \
    data.max_response_length=1024 \
    actor_rollout_ref.rollout.max_num_batched_tokens=26769 \
    data.filter_overlong_prompts=False \
    data.truncation='right' \
    custom_reward_function.path=/data1/reward_function.py \
    custom_reward_function.name="compute_score" \
    actor_rollout_ref.model.path=/data2/models/qwen2-7b-instruct \
    actor_rollout_ref.model.trust_remote_code=True \
    actor_rollout_ref.rollout.temperature=0.8 \
    actor_rollout_ref.rollout.top_p=0.95 \
    actor_rollout_ref.rollout.dtype=bfloat16 \
    actor_rollout_ref.actor.optim.lr=1e-6 \
    actor_rollout_ref.model.use_remove_padding=True \
    actor_rollout_ref.model.lora_rank=16 \
    actor_rollout_ref.model.lora_alpha=32 \
    actor_rollout_ref.model.target_modules=all-linear \
    actor_rollout_ref.actor.ppo_mini_batch_size=2 \
    actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu=1 \
    actor_rollout_ref.actor.use_kl_loss=True \
    actor_rollout_ref.actor.kl_loss_coef=0.001 \
    actor_rollout_ref.actor.kl_loss_type=low_var_kl \
    actor_rollout_ref.actor.entropy_coeff=0 \
    actor_rollout_ref.model.enable_gradient_checkpointing=True \
    actor_rollout_ref.rollout.enable_chunked_prefill=True \
    actor_rollout_ref.actor.fsdp_config.param_offload=True \
    actor_rollout_ref.actor.fsdp_config.optimizer_offload=True \
    actor_rollout_ref.rollout.log_prob_micro_batch_size_per_gpu=1 \
    actor_rollout_ref.rollout.tensor_model_parallel_size=4 \
    actor_rollout_ref.rollout.name=vllm \
    actor_rollout_ref.rollout.gpu_memory_utilization=0.95 \
    actor_rollout_ref.rollout.n=2 \
    actor_rollout_ref.rollout.load_format=safetensors \
    actor_rollout_ref.rollout.layered_summon=True \
    actor_rollout_ref.ref.log_prob_micro_batch_size_per_gpu=1 \
    actor_rollout_ref.ref.fsdp_config.param_offload=True \
    algorithm.use_kl_in_reward=False \
    trainer.critic_warmup=10 \
    trainer.logger=['console','tensorboard'] \
    trainer.log_val_generations=1 \
    trainer.project_name='grpo_longcite' \
    trainer.experiment_name='longcite_qwen2.5_7B' \
    trainer.n_gpus_per_node=4 \
    trainer.nnodes=1 \
    trainer.save_freq=100 \
    trainer.test_freq=100 \
    trainer.total_epochs=1 $@ > /data1//output_test.log 2>&1 &

<img width="1392" height="447" alt="Image" src="https://github.com/user-attachments/assets/b272c945-2c82-44c0-a4d6-799015e8cc4c" />

<img width="1113" height="59" alt="Image" src="https://github.com/user-attachments/assets/4c177ecf-7df2-4d7b-bfb3-a020e02e3fa0" />

### Expected behavior

解决

**Comment by ChicyChen** [comment]
请问你是怎么解决的？我在load local ckpt时出现了相同问题

**Comment by Eveosev** [comment]
同问

**Comment by C-myu** [comment]
同问

**Comment by C-myu** [comment]
export NCCL_P2P_DISABLE=1
单机多卡，把P2P禁用后解决了

**Comment by jianhai0527** [comment]
同问，我是在load reward model ckpt之后卡住，如何解决呢，多机多卡

**Comment by jianhai0527** [comment]
> 同问，我是在load reward model ckpt之后卡住，如何解决呢，多机多卡

换到main分支就好了

**Comment by SupreCyk** [comment]
> > 同问，我是在load reward model ckpt之后卡住，如何解决呢，多机多卡
> 
> 换到main分支就好了

请问能提供下您的main分支是哪一个提交记录么？

**Comment by jianhai0527** [comment]
> > > 同问，我是在load reward model ckpt之后卡住，如何解决呢，多机多卡
> > 
> > 
> > 换到main分支就好了
> 
> 请问能提供下您的main分支是哪一个提交记录么？

80eb57ea8a8fe8b9305865fc815b9e0ad5b30d14

**Comment by Kirrito-k423** [comment]
中文：

从目前公开信息看，这个卡住更像是多卡初始化阶段的 NCCL P2P / GPU 拓扑 / Ray worker 初始化问题，而不是单纯的 LoRA + FSDP + vLLM 配置不被支持。原因是：本 issue 下已有用户反馈单机多卡时 `export NCCL_P2P_DISABLE=1` 后解决；timeline 里也有外部提交把 `NCCL_P2P_DISABLE=1` 标注为 “verl #3445 workaround”；相似的多 GPU 初始化卡住 issue（例如 #3350、#597）也有多名用户反馈同一 workaround 有效。不过这通常会影响通信路径，可能降低训练速度。

建议先在启动脚本最前面尝试：

```bash
export NCCL_P2P_DISABLE=1
export HYDRA_FULL_ERROR=1
export RAY_DEDUP_LOGS=0
export NCCL_DEBUG=INFO
```

如果仍然卡住，或者禁用 P2P 后可以启动，请补充以下信息，方便判断是否需要 verl 侧修复、文档补充，还是环境/NCCL 拓扑问题：verl commit/version、PyTorch/CUDA/NCCL/driver/Ray/vLLM 版本、GPU 型号和数量、`nvidia-smi topo -m`、完整启动命令、完整文本日志（不要只贴截图）、`/tmp/ray/session_latest/logs/` 中相关 worker 日志，以及设置 `NCCL_P2P_DISABLE=1` 前后的对比结果。

English:

Based on the public evidence so far, this looks more like an NCCL P2P / GPU topology / Ray worker initialization issue during multi-GPU startup, rather than the LoRA + FSDP + vLLM configuration being unsupported in general. In this issue, one user reported that `export NCCL_P2P_DISABLE=1` fixed the single-node multi-GPU hang. The timeline also contains an external commit that explicitly records `NCCL_P2P_DISABLE=1` as a “verl #3445 workaround”. Similar multi-GPU initialization hangs in #3350 and #597 also have multiple reports that the same workaround helped. Note that disabling P2P can change the communication path and may slow training down.

Please try adding this before launching:

```bash
export NCCL_P2P_DISABLE=1
export HYDRA_FULL_ERROR=1
export RAY_DEDUP_LOGS=0
export NCCL_DEBUG=INFO
```

If it still hangs, or if disabling P2P makes it start, please share the verl commit/version, PyTorch/CUDA/NCCL/driver/Ray/vLLM versions, GPU model/count, `nvidia-smi topo -m`, the full launch command, full text logs instead of screenshots, relevant worker logs under `/tmp/ray/session_latest/logs/`, and a before/after comparison with `NCCL_P2P_DISABLE=1`.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
